core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -2971,7 +2981,29 @@ private void compileTruncate(SqlExecutionContext executionContext, @Transient Ch
                 if (Chars.isQuoted(tok)) {
                     tok = unquote(tok);
                 }
-                final TableToken tableToken = tableExistsOrFail(lexer.lastTokenPosition(), tok, executionContext);
+
+                final TableToken tableToken;
+                if (hasIfExists) {
+                    tableToken = executionContext.getTableTokenIfExists(tok);
+                    if (tableToken == null) {
+                        // Table doesn't exist, skip it when IF EXISTS is specified
+                        tok = SqlUtil.fetchNext(lexer);
+                        if (tok == null || Chars.equals(tok, ';') || isKeepKeyword(tok)) {
+                            break;
+                        }
+                        if (!Chars.equalsNc(tok, ',')) {
+                            throw SqlException.$(lexer.getPosition(), "',' or 'keep' expected");
```

Comment:
This branch is not covered by `TruncateTest`.

Reply:
fixed in https://github.com/questdb/questdb/pull/6020/commits/43058962bea79fb08df600b49ccde40d7259210c - there was unnecessary code duplication, copied from another PR :(